task/prefetch-dependencies/0.1/prefetch-dependencies.yaml

```diff
@@ -55,29 +72,70 @@ spec:
         dev_pacman_flag=""
       fi
 
+      if [ "$ENABLE_DEBUG" = "true" ]; then
+        debug_log_flag="--log-level=debug"
+      else
+        debug_log_flag=""
+      fi
+
+      if [ "${WORKSPACE_BASIC_AUTH_DIRECTORY_BOUND}" = "true" ] ; then
+        if [ -f "${WORKSPACE_BASIC_AUTH_DIRECTORY_PATH}/.git-credentials" ] && [ -f "${WORKSPACE_BASIC_AUTH_DIRECTORY_PATH}/.gitconfig" ]; then
+          cp "${WORKSPACE_BASIC_AUTH_DIRECTORY_PATH}/.git-credentials" "${PARAM_USER_HOME}/.git-credentials"
+          cp "${WORKSPACE_BASIC_AUTH_DIRECTORY_PATH}/.gitconfig" "${PARAM_USER_HOME}/.gitconfig"
+        # Compatibility with kubernetes.io/basic-auth secrets
+        elif [ -f "${WORKSPACE_BASIC_AUTH_DIRECTORY_PATH}/username" ] && [ -f "${WORKSPACE_BASIC_AUTH_DIRECTORY_PATH}/password" ]; then
+          HOSTNAME=$(echo $PARAM_URL | awk -F/ '{print $3}')
+          echo "https://$(cat ${WORKSPACE_BASIC_AUTH_DIRECTORY_PATH}/username):$(cat ${WORKSPACE_BASIC_AUTH_DIRECTORY_PATH}/password)@$HOSTNAME" > "${PARAM_USER_HOME}/.git-credentials"
+          echo -e "[credential \"https://$HOSTNAME\"]\n  helper = store" > "${PARAM_USER_HOME}/.gitconfig"
+        else
+          echo "Unknown basic-auth workspace format"
+          exit 1
+        fi
+        chmod 400 "${PARAM_USER_HOME}/.git-credentials"
+        chmod 400 "${PARAM_USER_HOME}/.gitconfig"
+        # needed or else you'll see "could not read Username for 'https://gitlab.com':"
+        cd $(workspaces.source.path)/source && git config remote.origin.url $(cat "${PARAM_USER_HOME}/.git-credentials")
```

Comment:
This doesn't seem right, the clone task doesn't need it. 

I think this approach may also print credentials in the task logs in some cases (IIUC that's why https://github.com/redhat-appstudio/build-definitions/pull/861 was needed, unfortunately can't tell from the anymore from the description)

Reply:
And cachi2 might include the credentials in the SBOM, it uses the origin url to construct the purl of packages coming from the source repo

Reply:
> And cachi2 might include the credentials in the SBOM, it uses the origin url to construct the purl of packages coming from the source repo

Filed https://github.com/containerbuildsystem/cachi2/issues/521

Reply:
I have altered my approach. please review